sympy/matrices/expressions/dotproduct.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import print_function, division
+
+from sympy.core import Basic, sympify
+from sympy.matrices.expressions.matexpr import (MatrixExpr)
+
+
+class DotProduct(MatrixExpr):
+    """
+        Dot Product of vector matrices
+    """
+
+    def __new__(cls, *args):
+        args = list(map(sympify, args))
+        if validate(*args):
+            return Basic.__new__(cls, *args)
+
+    @property
+    def args(self):
+        return self.args
+
+    def doit(self, expand=False):
+        try:
+            return dot_product(self.args)
+        except (AttributeError, NotImplementedError):
+            return self
+
+
+def dot_product(*args):
+    """
+    Calculates dot product of the arguments.
+    :param args: List of arguments for dot product.
+    :return: dot product of the arguments provided.
+    """
+    return True
```

Comment:
This should be filled out. All you need to have here is `(arg1*arg2)[0].doit()`. I also don't think this needs to be a separate function from `doit`.